### PR TITLE
Allow repo add to set authentication for private repository

### PR DESCRIPTION
**Pull request by Bao0ne**
Hi! Luet can access private repository by manually editing configuration files.I hope repo add can directly configure the authentication.

### CODE REVIEW

File: cmd/repo/add.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ Adds a repository to the system. URLs, local files or inline repo can be specifi
 
 # Inline:
 
- luet repo add testfo --description "Bar" --url "FOZZ" --type "ff"
+ luet repo add testfo --description "Bar" --url "FOZZ" --type "ff" --username "user" --passwd "password"
```
Comment (jimmykarily) [inline cmd/repo/add.go:50]: To demonstrate best practices, maybe this is better:

```suggestion
 luet repo add testfo --description "Bar" --url "FOZZ" --type "ff" --username "user" --passwd $(echo "$PASSWORD")
```

Generally, people shouldn't pass passwords in a way that it's written in bash history. Environment variables can be set in various ways, without running a command explicitly (which would have stored it in bash history too).

File: cmd/repo/add.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ Adds a repository to the system. URLs, local files or inline repo can be specifi
 
 # Inline:
```
Comment (jimmykarily) [inline cmd/repo/add.go:48]: Together with my comment below:

```suggestion
# Inline (provided you have $PASSWORD environent variable set):
```

**Review by jimmykarily — APPROVED** [review]
I wonder if it would be better if we read the docker authentication from the regular docker config (e.g. Using [`Load`](https://github.com/docker/cli/blob/master/cli/config/config.go#L122) and [`Dir`](https://github.com/docker/cli/blob/8454e012cadeca7719f170012c5331655ed5e5ec/cli/config/config.go#L73)) but I can't decide if we gain anything by that or not. Let's do it this way for now.


Just a small nit, the rest looks good.

**Review by jimmykarily — APPROVED** [review]

**Review by jimmykarily — APPROVED** [review]

**Review by mudler — APPROVED** [review]

**Comment by mudler** [comment]
Thank you!

**Pull request merged**